pkg/scheduler/framework/runtime/framework.go

```diff
@@ -88,6 +92,18 @@ type frameworkImpl struct {
 	parallelizer parallelize.Parallelizer
 }
 
+type instrumentedFilterPlugin struct {
```

Comment:
I think we can get rid of special slice by:
* changing filterPlugins back to []framework.FilterPlugin
* allowing getExtensionPoints to write there "raw" plugin
* later, in the same place as now you create `instrumentedFilterPlugins` slice, write (pseudocode) `i.filterPlugin[i] = &instrumentedFilterPlugin{i.filterPlugin[i], cachedMetric(i.filterPlugin[i]}`

instrumentedFilterPlugins can implement framework.FilterPlugin and increment the metric internally:

```suggestion
type instrumentedFilterPlugin struct {
	framework.FilterPlugin

	cachedMetric compbasemetrics.CounterMetric
}

func (i *instrumentedFilterPlugin) Filter(ctx context.Context, state *CycleState, pod *v1.Pod, nodeInfo *NodeInfo) *Status {
  i.cachedMetric.Inc()
  return i.instrumentedFilterPlugin.Filter(ctx, state, pod, nodeInfo)
}

var _ framework.FilterPlugin = &instrumentedFilterPlugin{}
```

Reply:
I'm happy with the above. Just make sure we don't report the metric increment if the plugin is supposed to be skipped.

Reply:
Thanks for the suggestion Maciej. Done.